pkg/kubelet/volumemanager/cache/desired_state_of_world.go

```diff
@@ -263,7 +263,7 @@ func (dsw *desiredStateOfWorld) AddPodToVolume(
 				if volumeSpec.Volume.EmptyDir != nil &&
 					volumeSpec.Volume.EmptyDir.SizeLimit != nil &&
 					volumeSpec.Volume.EmptyDir.SizeLimit.Value() > 0 &&
-					volumeSpec.Volume.EmptyDir.SizeLimit.Value() < sizeLimit.Value() {
+					(sizeLimit.Value() == 0 || volumeSpec.Volume.EmptyDir.SizeLimit.Value() < sizeLimit.Value()) {
```

Comment:
You've validated that this will work correctly if ephemeralStorageLimit is nil?

Reply:
yes, I have validated and you can see test case.

Reply:
@RobertKrawitz Hi, do you think there is any problem?

Reply:
/lgtm

I'm good with it.